packages/keybr-lesson/lib/text/fragment.ts

```diff
@@ -5,22 +5,23 @@ import { type WordGenerator } from "./words.ts";
 export function generateFragment(
   settings: Settings,
   nextWord: WordGenerator,
-  { doubleWords }: { readonly doubleWords: boolean },
+  { repeatWords }: { readonly repeatWords: number },
 ): string {
   const length = 100 + Math.round(settings.get(lessonProps.length) * 100);
   const words: string[] = [];
   let wordsLength = 0;
+  const occurences = repeatWords + 1;
+
   while (true) {
     const word = nextWord() || "?";
-    words.push(word);
-    wordsLength += word.length;
-    if (doubleWords) {
+
+    for (let i = 0; i < occurences; i++) {
       words.push(word);
       wordsLength += word.length;
-    }
-    if (wordsLength >= length) {
-      break;
+
+      if (wordsLength >= length) {
+        return words.join(" ");
+      }
```

Comment:
Now that length is variable, should we allow words at the end not to repeat? Currently that's what's been implemented.

Reply:
Wdym?
The algorithm will fill as many words as possible to reach the target length.
It will also truncate a repeated words sequence if it overflows the target length.
I think this is the correct behavior.

Reply:
Beforehand it wasn't truncating.
It would always ensure there are two words even if it goes above the word length.